src/main/java/thaumicboots/api/ItemBoots.java

```diff
@@ -292,17 +292,18 @@ public static void renderHUDSpeedNotification() {
         Minecraft mc = Minecraft.getMinecraft();
         String text = getModeText(
                 "thaumicboots.speedEffect",
-                getBoots(mc.thePlayer).stackTagCompound.getDouble(TAG_MODE_SPEED) * 100);
+                IBoots.getBoots(mc.thePlayer).stackTagCompound.getDouble(TAG_MODE_SPEED) * 100);
```

Comment:
same here

Reply:
getBoots() returns null with non ItemBoots item (EMT boots)

This will cause a clientside NPE crash when changing speed.

Reply:
It absolutely shouldn't?

Reply:
```
public static ItemStack getBoots(EntityPlayer player) {
        ItemStack stack1 = player.getCurrentArmor(0);
        return isBoot(stack1) ? stack1 : null;
    }
```

IsBoot is false, will return null for EMT boots.

Reply:
Lmfao that's some kooky shit

Reply:
I'll take a look at the code in  a bit

Reply:
fixed locally about to push